src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java

```diff
@@ -300,6 +307,53 @@ private void testAddQuestionAction() throws Exception {
         feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackQuestionAddSuccess.html");
     }
 
+    private void testAddTemplateQuestionLink() throws Exception {
+
+        ______TS("add template question link");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.clickTemplateModalButton();
+
+        feedbackEditPage.verifyHtmlPart(By.id("addTemplateQuestionModal"),
+                "/instructorFeedbackTemplateQuestionExpandedModal.html");
+
+        assertFalse("Should not be able to submit when there is no question selected",
+                feedbackEditPage.isAddTemplateQuestionButtonEnabled());
+    }
+
+    private void testAddTemplateQuestionAction() throws Exception {
+
+        ______TS("add template question action success");
+
+        // check if each question panel is expandable
+        feedbackEditPage.clickTemplateQuestionPanel(1);
+        assertTrue(feedbackEditPage.isTemplateQuestionPanelExpanded(1));
+
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(1);
+
+        assertTrue("Can click after selecting a question",
+                feedbackEditPage.isAddTemplateQuestionButtonEnabled());
+        feedbackEditPage.clickAddTemplateQuestionButton();
+
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+        assertNotNull(getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 1));
+
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackTemplateQuestionAddSuccess.html");
+
+        ______TS("add multiple template questions success");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.clickTemplateModalButton();
+
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(2);
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(3);
+        feedbackEditPage.clickAddTemplateQuestionButton();
+
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(
+                Const.StatusMessages.FEEDBACK_QUESTION_ADDED_MULTIPLE);
+        assertNotNull(getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 3));
```

Comment:
Made the necessary changes for everything except this. Currently (local and travis), `testTemplateQuestionAdd()` test can pass on its own but if it runs after `allTests()` in `InstructorFeedbackEditPageUiTest`.

It is failing due to the difference in html seen as `allTests()` ends with Feedback Session being "Open Session" while `testTemplateQuestionAdd()` initialises the session as "First Feedback Session", causing the difference in verifying of the UI.

I am not really familiar with how to deal with the situation. I feel that maybe we should add a method to `testTemplateQuestionAdd()` so that it will use the same feedbackSession and pass when it runs together or runs separately. What do you think?

Reply:
```java
@Override
protected void prepareTestData() {
    // see prepareData()
}

@BeforeMethod
protected prepareData() {
    // all content in original prepareTestData()
    // move classSetup() code also to here and remove classSetup() method
}
```

In doing this. We are saying before every `@Test` in this classes, `prepareData()` must be run first()

Reply:
Works really well! 👍  I believe by doing this, we do not need to remove any questions being added at the end anymore since they wont affect other test cases?

Reply:
Yes.